### PR TITLE
MINIFICPP-2154 Replace SecureSocketGetTCPTest with utils::net::getSSLContext tests

### PR DESCRIPTION
**Pull request by martinzink**
This will also enable us to remove SecureSocketGetTCPTest after MINIFICPP-2131

MINIFICPP-2154 Replace SecureSocketGetTCPTest with utils::net::getSSLContext tests

---
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/CMakeLists.txt [hunk 1]
```diff
@@ -78,7 +78,7 @@ endif()
 file(GLOB PROCESSOR_SOURCES  "src/processors/*.cpp" )
 
 
-add_library(core-minifi SHARED ${SOURCES})
+add_library(core-minifi SHARED ${SOURCES} include/utils/net/AsioSocketUtils.h)
```
Comment (lordgamez) [inline libminifi/CMakeLists.txt:81]: Is include/utils/net not included in the include directories or why is it needed to be added to the sources?

File: cmake/BuildTests.cmake [hunk 2]
```diff
@@ -30,6 +30,16 @@ ENDMACRO()
 
 set(NANOFI_TEST_DIR "${CMAKE_SOURCE_DIR}/nanofi/tests/")
 
+function(configureTestResources SOURCE_DIR DEST_DIR)
```
Comment (lordgamez) [inline cmake/BuildTests.cmake:33]: As it doesn't do any configuration maybe it should be named something like `copyTestResources`

File: libminifi/CMakeLists.txt (hunk 1 above)
Comment (martinzink) [reply libminifi/CMakeLists.txt]: that was probably my IDE automatically adding it, thanks for catching. removed it in https://github.com/apache/nifi-minifi-cpp/pull/1598/commits/3d451e15619339b1f54d65ee31f3c000a4f4e25f#diff-384b8bf2d8b6601d04935ea6bc051dee880b128f5add4fc99b66ca0f6a7bccbaL81

**Review by lordgamez — APPROVED** [review]

**Comment by szaszm** [comment]
What caused this reading from stdin and which part fixes it? I'm only seeing renames, moving stuff around, and added client cert handling to the test.

**Comment by martinzink** [comment]
> What caused this reading from stdin and which part fixes it? I'm only seeing renames, moving stuff around, and added client cert handling to the test.

Yeah it looks like my GetTCP PR got merged so that already fixed the issue.
The problem was that `use_private_key_file` was called before `set_password_callback` and `use_private_key_file` calls the password callback https://github.com/apache/nifi-minifi-cpp/commit/68ba80a9d135c0783c369c848448c6c52401c170#diff-a62ddb0368e1813dff2a90fe7433e148bfc7b21b7d7ad9c8a066fd040d209ac9L125-L126

The rebase made this change not really a fix only tests so I'll rename the PR.

**Review by adamdebreceni — APPROVED** [review]

File: libminifi/test/unit/NetUtilsTest.cpp [hunk 3]
```diff
@@ -64,3 +65,44 @@ TEST_CASE("net::reverseDnsLookup", "[net][dns][reverseDnsLookup]") {
...
 }
+
+TEST_CASE("utils::net::getClientSslContext") {
+  TestController controller;
+  auto plan = controller.createPlan();
+
+  auto ssl_context_node = plan->addController("SSLContextService", "ssl_context_service");
+  auto ssl_context_service = std::dynamic_pointer_cast<minifi::controllers::SSLContextService>(ssl_context_node->getControllerServiceImplementation());
+
+  const std::filesystem::path ca_dir = std::filesystem::path(minifi::utils::file::FileUtils::get_executable_dir()) / "resources";
```
[the remark this answers is not in the document]
Comment (martinzink) [reply libminifi/test/unit/NetUtilsTest.cpp]: good idea, changed it to cert_dir in https://github.com/apache/nifi-minifi-cpp/pull/1598/commits/4520d969e058763ccee42197547b9993c8604f86#diff-a383222e204712d33628e58bf0e17c46f8480d366677941aaca4c4743747126fL76

File: libminifi/test/unit/NetUtilsTest.cpp [hunk 4]
```diff
@@ -64,3 +65,44 @@ TEST_CASE("net::reverseDnsLookup", "[net][dns][reverseDnsLookup]") {
     CHECK(unresolvable_hostname == "2001:db8::");
   }
 }
+
+TEST_CASE("utils::net::getClientSslContext") {
```
[the remark this answers is not in the document]
Comment (martinzink) [reply libminifi/test/unit/NetUtilsTest.cpp:69]: Added some tests that deal with passphrase problems and one that deals with missing CA the later one also revealed a problem (CA is not required in SSLContextService, so we should check if its there before setting into the asio ssl_context) https://github.com/apache/nifi-minifi-cpp/pull/1598/commits/4520d969e058763ccee42197547b9993c8604f86#diff-a383222e204712d33628e58bf0e17c46f8480d366677941aaca4c4743747126fR110-R158

**Pull request closed without merging**